src/components/transactions/HexEncodedData/index.tsx

```diff
@@ -18,22 +19,29 @@ export const HexEncodedData = ({ hexData, title, limit = 20 }: Props): ReactElem
     setShowTxData((val) => !val)
   }
 
+  const firstBytes = <b>{hexData.slice(0, 8)}</b>
+  const restBytes = hexData.slice(8)
+
   return (
     <Box data-testid="tx-hexData" className={css.encodedData}>
       {title && (
         <span>
           <b>{title}: </b>
         </span>
       )}
+
+      <CopyButton text={hexData} />
+
+      {firstBytes}
       {showExpandBtn ? (
         <>
-          {showTxData ? hexData : shortenText(hexData, 25)}{' '}
+          {showTxData ? restBytes : shortenText(restBytes, 25)}{' '}
```

Comment:
As we now always show the first 10 characters separately we have to shorten more.
```suggestion
          {showTxData ? restBytes : shortenText(restBytes, 15)}{' '}
```